File: C++/1. 1304. Find N Unique Integers Sum up to Zero.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> res;
        for(int i = 1; i <= n / 2; i++)
        {
            res.push_back(i);
            res.push_back(-i);
        }
        if(n % 2 != 0)
        {
            res.push_back(0);
        }
        return res;
    }
};

int main() {
    Solution s;
    vector<int> result = s.sumZero(5);
    cout << "Result for n = 5: ";
    for(int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    cout << endl;

    result = s.sumZero(3);
    cout << "Result for n = 3: ";
    for(int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    cout << endl;

    result = s.sumZero(1);
    cout << "Result for n = 1: ";
    for(int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    cout << endl;

    return 0;
}

File: Python/1. 1304. Find N Unique Integers Sum up to Zero.py
class Solution:
    def sumZero(self, n: int) -> List[int]:
        if n == 1:
            return [0]
        res = []
        for i in range(1, n//2 + 1):
            res.append(i)
            res.append(-i)
        if n % 2 != 0:
            res.append(0)
        return res
s = Solution()
n = 3
result = []
result = s.sumZero(n);
for i in range(len(result)):
    print(result[i])
